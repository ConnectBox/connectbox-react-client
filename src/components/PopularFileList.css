td.popular-content-header {
    border-top: 0px !important;
    background-color: #CCEFEB;
    border-radius: 4px;
}
