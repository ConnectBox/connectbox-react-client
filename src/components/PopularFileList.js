import './FolderList.css'
import './PopularFileList.css'

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import PopularFileListEntry from './PopularFileListEntry'

class PopularFileList extends Component {
    static propTypes = {
      popularFiles: PropTypes.array.isRequired
    }

  render () {
    const { popularFiles } = this.props
    const popularFileElements = []
    for (let i = 0; popularFiles && i < popularFiles.length; i++) {
      const file = popularFiles[i]
      
      popularFileElements.push(
        <PopularFileListEntry
          key={`pop${i}`}
          file={file}
          hideTopBorder={i === 0}
        />
      )
    }

    return (
      <div className="table-responsive">
        <table className="table">
            <tbody>
                <tr><td className="popular-content-header"><i className="fa fa-line-chart"></i></td></tr>
                {popularFileElements}
            </tbody>
        </table>
      </div>
    )
  }
}

export default PopularFileList
