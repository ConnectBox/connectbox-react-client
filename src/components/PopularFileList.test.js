/* global it, expect, */
import React from 'react'
import PopularFileList from './PopularFileList'
import renderer from 'react-test-renderer'

it('Render PopularFileList', () => {
  const iconMetadata = {ids: [], names: {}}
  const content = [{type: 'file', ext: 'doc', name: 'Test File'}]
  const component = renderer.create(
    <PopularFileList
      contentRoot='/content'
      content={content}
      folderPath='foo'
      iconMetadata={iconMetadata}
      popularFiles={content} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
