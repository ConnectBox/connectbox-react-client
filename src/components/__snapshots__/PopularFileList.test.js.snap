// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PopularFileList 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td
          className="popular-content-header"
        >
          <i
            className="fa fa-line-chart"
          />
        </td>
      </tr>
      <tr>
        <td
          style={
            Object {
              "borderTop": 0,
            }
          }
        >
          <a
            href={undefined}
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <i
              className="fa fa-file-text-o"
            />
            <span>
               
              Test File
            </span>
          </a>
          <span
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
             
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
